### DISCUSSION TITLE
unable to load "glb" models from device storage path or Asset path

### DISCUSSION THREAD

**Discussion by nayash** [google/model-viewer#3749] [Q&A]
I want to use the Android Scene Viewer in Android by passing following intent:

```
val sceneViewerIntent = Intent(Intent.ACTION_VIEW)
val intentUri: Uri = Uri.parse("https://arvr.google.com/scene-viewer/1.0").buildUpon()
    .appendQueryParameter(
        "file:///android_asset/Avocado.glb"
        // file.absolutePath // even a file in app's files dir doesn't work
        // "https://raw.githubusercontent.com/KhronosGroup/glTF-Sample-Models/master/2.0/Avocado/glTF-Binary/Avocado.glb" // this works
    )
    .appendQueryParameter("mode", "3d_preferred")
    .build()
sceneViewerIntent.data = intentUri
sceneViewerIntent.setPackage("com.google.ar.core")
startActivity(sceneViewerIntent)
```
I keep getting "Couldn't load object, looks like there's something wrong with this object". But nothing is wrong with the object because I'm able to load it in https://modelviewer.dev/editor/.
 
Although URL works, I want to download the glb file first and then pass it via intent because I want the assets/resources to be fetched via authenticated APIs (which I'm not sure if possible via intent URLs like this). So my questions are:

* How to load local "glb" files (from app's internal storage directory)?
* If above is not possible, how can I protect my "glb" files behind an authentication layer, so that only authenticated users can view the models?

Similar discussions are still unanswered since a long time:
https://github.com/google/model-viewer/discussions/2187
https://github.com/google/model-viewer/discussions/3680 (no answers)

**Comment by elalish** [comment]
I don't think SceneViewer is set up for either of those bullets - @eugenecys can you verify? Not sure if there's a work-around.

**Reply by nayash** [reply]
Hey @elalish / @eugenecys , any hints/solutions for this? I can understand that auth may not be straightforward but why is local model paths not allowed? internally the scene viewer must be already downloading & saving the model before rendering it right? Would really appreciate a resolution for this.
